### PR TITLE
Use dark titlebar if dark mode is enabled

### LINKED ISSUE #90: Use dark GTK theme like the the Spotify flatpak
Spotify on flatpak uses dark GTK theme which looks nicer than the default theme on GNOME. Perhaps the flatpak for Discord should do the same? any opinions? 

Screenshot from 2020-03-27 08-06-05

### PR DESCRIPTION
**Pull request by connlim**
Fixes #90 

Based on the [script used by the Spotify flatpak](https://github.com/flathub/com.spotify.Client/blob/master/set-dark-theme-variant.py).

Runs a python script in the background when launching Discord. The script sets a dark titlebar if the Discord config `BACKGROUND_COLOR` is `#202255`. (see https://github.com/flathub/com.discordapp.Discord/issues/90#issuecomment-695822104). The script will attempt to set the titlebar color for 5 seconds, and defaults to a fallback if it fails to do so within that timeframe. 

Limitation: because the script is only run on app startup, the titlebar theme will not be updated until the app is restarted

### CODE REVIEW

File: set-gtk-dark-theme.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            else:
+                fallback(window_titles, variant)
+                break
+
+
+if __name__ == "__main__":
+    xdg_config_home = os.environ.get("XDG_CONFIG_HOME") or os.path.join(
+        os.path.expanduser("~"), ".config"
+    )
+    with open(
```
Comment (TingPing) [inline set-gtk-dark-theme.py:155]: All of this should be robust against exceptions. Opening the file, parsing the json, etc.

Comment (connlim) [reply set-gtk-dark-theme.py]: Thanks for pointing that out. I've added a try block to the script that should cover the most common exceptions.

**Comment by lionirdeadman** [comment]
It's not my area of expertise but is using X11 to change the window decoration optimal? This is going to break when Discord will use Wayland, right?

(Also this needs to be updated for the 0.13 changes, sorry about that)

**Comment by TingPing** [comment]
> This is going to break when Discord will use Wayland, right?

That's a discussion that's probably a year down the road but on Wayland its up to clients to draw their decorations and *hopefully* at that point Discord will render its own buttons like it does on all other platforms and the workaround will no longer be required.

At that point calling the X apis will just harmlessly fail. It won't silently happen though because this package doesn't have wayland permissions so somebody will test it.

**Comment by lionirdeadman** [comment]
> That's a discussion that's probably a year (or more) down the road but on Wayland its up to clients to draw their decorations and _hopefully_ at that point Discord will render its own buttons like it does on all other platforms and the workaround will no longer be required.

I was under the impression it already did but it seems it uses server-side decorations so trying to make it use something else than X would be useless until we know what they end up doing, you're right.

> At that point calling the X apis will just harmlessly fail. It won't silently happen though because this package doesn't have wayland permissions so somebody will test it.

That's fair though. It works.

Just need the 0.13 change and then I'll merge it then.

**Comment by connlim** [comment]
Hi will this be merged in any time soon? I see that this repo has already been updated to version 0.13. Thanks.

**Comment by Fuseteam** [comment]
@connlim this repo has yes but iiuc the merge is waiting for you to update the pr. there is a merge conflict due to the 0.13 update for example

**Comment by connlim** [comment]
Oops my bad. I've just fixed the merge conflicts. Please take a look again.

**Comment by Fuseteam** [comment]
thanks @connlim guess now we wait till @lionirdeadman has taken a look

**Comment by lionirdeadman** [comment]
Looks like it works, I see nothing wrong with it. Thanks for working on this!

**Comment by alexmex90** [comment]
Looks amazing and works perfectly, feels good to see something I suggested materialized like this, thank you very much!

**Pull request merged**